### ISSUE TITLE
[SDPA] iree-compile aborts on SDPA with head_dim=1 for rocm/gfx942

### ISSUE THREAD

**Issue by rkayaith** [iree-org/iree#24213]
`iree-compile` aborts on `torch.aten.scaled_dot_product_attention` with `head_dim=1` for the rocm/gfx942 target.

## Repro
This IR is generated from hipDNN/fusili as part of the cross-attention layer in the `Transformer` decoder in this test: https://github.com/pytorch/pytorch/blob/6af13031e55a1f824b4cabe23b733e8ad4e9a24c/test/test_transformers.py#L1511

`sdpa_head_dim_1.mlir`:

```mlir
module @module {
  func.func @main(%o_3_: !torch.tensor<[1,2,4,1],f16>, %k_1: !torch.vtensor<[1,2,4,1],f16>, %q_0: !torch.vtensor<[1,2,4,1],f16>, %v_2: !torch.vtensor<[1,2,4,1],f16>) attributes {torch.assume_strict_symbolic_shapes} {
    %permute_Q_val_0_sdpa_fprop = torch.constant.int 0
    %permute_Q_val_1_sdpa_fprop = torch.constant.int 1
    %permute_Q_val_2_sdpa_fprop = torch.constant.int 2
    %permute_Q_val_3_sdpa_fprop = torch.constant.int 3
    %permute_Q_sdpa_fprop = torch.prim.ListConstruct %permute_Q_val_0_sdpa_fprop, %permute_Q_val_1_sdpa_fprop, %permute_Q_val_2_sdpa_fprop, %permute_Q_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %q_0_sdpa_fprop_perm = torch.aten.permute %q_0, %permute_Q_sdpa_fprop : !torch.vtensor<[1,2,4,1],f16>, !torch.list<int> -> !torch.vtensor<[1,2,4,1],f16>

    %permute_K_val_0_sdpa_fprop = torch.constant.int 0
    %permute_K_val_1_sdpa_fprop = torch.constant.int 1
    %permute_K_val_2_sdpa_fprop = torch.constant.int 2
    %permute_K_val_3_sdpa_fprop = torch.constant.int 3
    %permute_K_sdpa_fprop = torch.prim.ListConstruct %permute_K_val_0_sdpa_fprop, %permute_K_val_1_sdpa_fprop, %permute_K_val_2_sdpa_fprop, %permute_K_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %k_1_sdpa_fprop_perm = torch.aten.permute %k_1, %permute_K_sdpa_fprop : !torch.vtensor<[1,2,4,1],f16>, !torch.list<int> -> !torch.vtensor<[1,2,4,1],f16>

    %permute_V_val_0_sdpa_fprop = torch.constant.int 0
    %permute_V_val_1_sdpa_fprop = torch.constant.int 1
    %permute_V_val_2_sdpa_fprop = torch.constant.int 2
    %permute_V_val_3_sdpa_fprop = torch.constant.int 3
    %permute_V_sdpa_fprop = torch.prim.ListConstruct %permute_V_val_0_sdpa_fprop, %permute_V_val_1_sdpa_fprop, %permute_V_val_2_sdpa_fprop, %permute_V_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %v_2_sdpa_fprop_perm = torch.aten.permute %v_2, %permute_V_sdpa_fprop : !torch.vtensor<[1,2,4,1],f16>, !torch.list<int> -> !torch.vtensor<[1,2,4,1],f16>

    %none_mask_sdpa_fprop = torch.constant.none
    %dropout_sdpa_fprop = torch.constant.float 0.000000e+00
    %is_causal_sdpa_fprop = torch.constant.bool false
    %scale_sdpa_fprop = torch.constant.float 1.000000e+00
    %enable_gqa_sdpa_fprop = torch.constant.bool false
    %o_3_sdpa_fprop_perm = torch.aten.scaled_dot_product_attention %q_0_sdpa_fprop_perm, %k_1_sdpa_fprop_perm, %v_2_sdpa_fprop_perm, %none_mask_sdpa_fprop, %dropout_sdpa_fprop, %is_causal_sdpa_fprop, %scale_sdpa_fprop, %enable_gqa_sdpa_fprop : !torch.vtensor<[1,2,4,1],f16>, !torch.vtensor<[1,2,4,1],f16>, !torch.vtensor<[1,2,4,1],f16>, !torch.none, !torch.float, !torch.bool, !torch.float, !torch.bool -> !torch.vtensor<[1,2,4,1],f16>

    %permute_O_val_0_sdpa_fprop = torch.constant.int 0
    %permute_O_val_1_sdpa_fprop = torch.constant.int 1
    %permute_O_val_2_sdpa_fprop = torch.constant.int 2
    %permute_O_val_3_sdpa_fprop = torch.constant.int 3
    %permute_O_sdpa_fprop = torch.prim.ListConstruct %permute_O_val_0_sdpa_fprop, %permute_O_val_1_sdpa_fprop, %permute_O_val_2_sdpa_fprop, %permute_O_val_3_sdpa_fprop : (!torch.int, !torch.int, !torch.int, !torch.int) -> !torch.list<int>
    %o_3 = torch.aten.permute %o_3_sdpa_fprop_perm, %permute_O_sdpa_fprop : !torch.vtensor<[1,2,4,1],f16>, !torch.list<int> -> !torch.vtensor<[1,2,4,1],f16>

    torch.overwrite.tensor.contents %o_3 overwrites %o_3_ : !torch.vtensor<[1,2,4,1],f16>, !torch.tensor<[1,2,4,1],f16>
    return
  }
}
```

```bash
iree-compile sdpa_head_dim_1.mlir --iree-hal-target-backends=rocm --iree-rocm-target=mi300x
```

## Crash output

```
iree-compile: ../third_party/llvm-project/llvm/include/llvm/ADT/ArrayRef.h:152: const T &llvm::ArrayRef<long>::back() const [T = long]: Assertion `!empty()' failed.
Please report issues to https://github.com/iree-org/iree/issues and include the crash backtrace.
Stack dump:
0.	Program arguments: iree/build/tools/iree-compile tmp/iree_segfault_repro/sdpa_head_dim_1.mlir --iree-hal-target-backends=rocm --iree-rocm-target=mi300x -o tmp/out.vmfb
 #0 0x0000711a1cf6afc8 llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) iree/build/../third_party/llvm-project/llvm/lib/Support/Unix/Signals.inc:880:13
 #1 0x0000711a1cf68870 llvm::sys::RunSignalHandlers() iree/build/../third_party/llvm-project/llvm/lib/Support/Signals.cpp:109:18
 #2 0x0000711a1cf6bd61 SignalHandler(int, siginfo_t*, void*) iree/build/../third_party/llvm-project/llvm/lib/Support/Unix/Signals.inc:448:38
 #3 0x0000711a15842520 (/lib/x86_64-linux-gnu/libc.so.6+0x42520)
 #4 0x0000711a158969fc __pthread_kill_implementation ./nptl/./nptl/pthread_kill.c:44:76
 #5 0x0000711a158969fc __pthread_kill_internal ./nptl/./nptl/pthread_kill.c:78:10
 #6 0x0000711a158969fc pthread_kill ./nptl/./nptl/pthread_kill.c:89:10
 #7 0x0000711a15842476 gsignal ./signal/../sysdeps/posix/raise.c:27:6
 #8 0x0000711a158287f3 abort ./stdlib/./stdlib/abort.c:81:7
 #9 0x0000711a1582871b _nl_load_domain ./intl/./intl/loadmsgcat.c:1177:9
#10 0x0000711a15839e96 (/lib/x86_64-linux-gnu/libc.so.6+0x39e96)
#11 0x0000711a1f680a0a mlir::iree_compiler::IREE::LinalgExt::detail::OnlineAttentionOpGenericAdaptorBase::Properties* mlir::PropertyRef::as<mlir::iree_compiler::IREE::LinalgExt::detail::OnlineAttentionOpGenericAdaptorBase::Properties*>() const iree/build/../third_party/llvm-project/mlir/include/mlir/IR/OperationSupport.h:83:5
#12 0x0000711a1f680a0a mlir::PropertiesSelector<mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp, void>::type& mlir::Op<mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp, mlir::OpTrait::OneRegion, mlir::OpTrait::VariadicResults, mlir::OpTrait::ZeroSuccessors, mlir::OpTrait::AtLeastNOperands<7u>::Impl, mlir::OpTrait::SingleBlock, mlir::OpTrait::SingleBlockImplicitTerminator<mlir::iree_compiler::IREE::LinalgExt::YieldOp>::Impl, mlir::OpTrait::OpInvariants, mlir::BytecodeOpInterface::Trait, mlir::iree_compiler::IREE::LinalgExt::LinalgFusionOpInterface::Trait, mlir::IndexingMapOpInterface::Trait, mlir::ReifyRankedShapedTypeOpInterface::Trait, mlir::linalg::AggregatedOpInterface::Trait, mlir::TilingInterface::Trait, mlir::PartialReductionOpInterface::Trait, mlir::MemoryEffectOpInterface::Trait, mlir::DestinationStyleOpInterface::Trait, mlir::iree_compiler::IREE::LinalgExt::LinalgExtOp::Trait>::getProperties<mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp>() iree/build/../third_party/llvm-project/mlir/include/mlir/IR/OpDefinition.h:1983:27
#13 0x0000711a1f680a0a mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp::setDecompositionConfigAttr(mlir::DictionaryAttr) iree/build/compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.h.inc:4495:5
#14 0x0000711a1f680a0a mlir::iree_compiler::setAttentionReductionConfig(mlir::iree_compiler::AttentionReductionHeuristicSeeds&, mlir::iree_compiler::IREE::GPU::TargetAttr, mlir::FunctionOpInterface, mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp) iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1356:6
#15 0x0000711a1f675118 mlir::iree_compiler::setAttentionVectorDistributionConfig(mlir::iree_compiler::IREE::GPU::TargetAttr, mlir::FunctionOpInterface, mlir::iree_compiler::IREE::LinalgExt::OnlineAttentionOp) iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1428:10
#16 0x0000711a1f675118 mlir::iree_compiler::setVectorDistributionConfig(mlir::iree_compiler::IREE::GPU::TargetAttr, mlir::FunctionOpInterface, mlir::Operation*) iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:1464:12
#17 0x0000711a1f66e35e mlir::iree_compiler::setRootConfig(mlir::iree_compiler::IREE::GPU::TargetAttr, mlir::FunctionOpInterface, mlir::Operation*) iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:2432:7
#18 0x0000711a1f66d99f mlir::iree_compiler::initGPULaunchConfig(mlir::FunctionOpInterface) iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp:2675:7
#19 0x0000711a1f66bbd0 mlir::iree_compiler::(anonymous namespace)::LLVMGPUSelectLoweringStrategyPass::runOnOperation() iree/build/../compiler/src/iree/compiler/Codegen/LLVMGPU/LLVMGPUSelectLoweringStrategy.cpp:84:9
...
```

## Environment

- IREE built from source @ https://github.com/iree-org/iree/commit/00dc51c9bf143cbdb23395fd1704988e9e659a18

**Comment by IanWood1** [comment]
Looks like it's an issue in dispatch creation. When head_dim = 1, we fold away the head dim completely which causes the crash in codegen. I'll create a fix for this similar to https://github.com/iree-org/iree/pull/24127

**Comment by Groverkss** [comment]
@IanWood1 I'm going to assume you have this on handle. Let me know if you need me to have a look.
